pkg/controller/statefulset/stateful_set.go

```diff
@@ -130,7 +130,7 @@ func NewStatefulSetController(
 	ssc.podLister = podInformer.Lister()
 	ssc.podListerSynced = podInformer.Informer().HasSynced
 
-	setInformer.Informer().AddEventHandler(
+	setInformer.Informer().AddEventHandlerWithResyncPeriod(
```

Comment:
It seems so. But I am curious about one thing, will there be any problem with using the default one? Because the controller has a default synchronization period which seems to be a better choice. In addition, as mentioned in this comment, other controllers have the same settings. https://github.com/kubernetes/kubernetes/pull/75622#issuecomment-475867919

Reply:
The default resync period is between 12 and 24 hours, and it's still possible to cause the problem  in #https://github.com/kubernetes/kubernetes/issues/75495. In this PR, the resync is disabled completely for sts controller to avoid the problem.

Reply:
The default sync time seems to be a good way for controllers because the controller mode requires some time to synchronize the original resources for reconciliation. This seems to be both the advantage and the original intention of this design pattern. The answer to this question is also answered in this comment.https://github.com/kubernetes/kubernetes/pull/75622#issuecomment-475867919  So, I'm curious if you have encountered the same issue due to this default sync time, and if not, I think sticking with the default is a better choice.

Reply:
It seems that the issue encountered at that time was likely due to the short synchronization period (30 seconds), which could cause similar problems when dealing with a large number of pods. If the default sync time (between 12 and 24 hours) is used, I believe that synchronizing once in such a long cycle would not lead to this issue. (Please correct me if my understanding is incorrect.)